platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/platformstatus/statuslogic/ReplayingEventsStatusLogic.java

```diff
@@ -0,0 +1,206 @@
+
+import com.swirlds.common.system.platformstatus.IllegalPlatformStatusException;
+import com.swirlds.common.system.platformstatus.PlatformStatus;
+import com.swirlds.common.system.platformstatus.PlatformStatusConfig;
+import com.swirlds.common.system.platformstatus.statusactions.CatastrophicFailureAction;
+import com.swirlds.common.system.platformstatus.statusactions.DoneReplayingEventsAction;
+import com.swirlds.common.system.platformstatus.statusactions.FallenBehindAction;
+import com.swirlds.common.system.platformstatus.statusactions.FreezePeriodEnteredAction;
+import com.swirlds.common.system.platformstatus.statusactions.ReconnectCompleteAction;
+import com.swirlds.common.system.platformstatus.statusactions.SelfEventReachedConsensusAction;
+import com.swirlds.common.system.platformstatus.statusactions.StartedReplayingEventsAction;
+import com.swirlds.common.system.platformstatus.statusactions.StateWrittenToDiskAction;
+import com.swirlds.common.system.platformstatus.statusactions.TimeElapsedAction;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Objects;
+
+/**
+ * Class containing the state machine logic for the {@link PlatformStatus#REPLAYING_EVENTS} status.
+ */
+public class ReplayingEventsStatusLogic implements PlatformStatusLogic {
+    /**
+     * The platform status config
+     */
+    private final PlatformStatusConfig config;
+
+    /**
+     * The round number of the freeze period if one has been entered, otherwise null
+     */
+    private Long freezeRound = null;
+
+    /**
+     * Constructor
+     *
+     * @param config the platform status config
+     */
+    public ReplayingEventsStatusLogic(@NonNull final PlatformStatusConfig config) {
+        this.config = Objects.requireNonNull(config);
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * {@link PlatformStatus#REPLAYING_EVENTS} status unconditionally transitions to
+     * {@link PlatformStatus#CATASTROPHIC_FAILURE} when a {@link CatastrophicFailureAction} is processed.
+     */
+    @NonNull
+    @Override
+    public PlatformStatusLogic processCatastrophicFailureAction(@NonNull final CatastrophicFailureAction action) {
+        return new CatastrophicFailureStatusLogic();
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * If a freeze boundary wasn't crossed while replaying events, then receiving a {@link DoneReplayingEventsAction}
+     * causes a transition to {@link PlatformStatus#OBSERVING}.
+     * <p>
+     * If a freeze boundary was crossed while replaying events, then the {@link DoneReplayingEventsAction} doesn't
+     * affect the state machine. The status will remain in {@link PlatformStatus#REPLAYING_EVENTS} until the freeze
+     * state has been saved.
+     */
+    @NonNull
+    @Override
+    public PlatformStatusLogic processDoneReplayingEventsAction(@NonNull final DoneReplayingEventsAction action) {
+        // always transition to a new status when done replaying events
```

Comment:
comment does not match the implementation

Reply:
Removed old comment